Build failure with newest zig master due to new alignForward definition

### Zig Version

0.11.0-dev.3681+d41111d7e

### Zig Language Server Version

f6da47fafdbc89ab628613fea52897c720b96564

### Steps to Reproduce

Build zls with the specified version of Zig, with `zig build -Doptimize=ReleaseSafe`.

### Expected Behavior

I'd expect the build to succeed. (Not much to say here.)

### Actual Behavior

The build fails and following error message is printed:
```
Build Summary: 1/4 steps succeeded; 1 failed (disable with --summary none)
install transitive failure
└─ install zls transitive failure
   └─ zig build-exe zls ReleaseSafe native 1 errors
src/analyser/encoding.zig:196:27: error: expected 3 argument(s), found 2
    const offset = std.mem.alignForward(unaligned, alignment) - unaligned;
                   ~~~~~~~^~~~~~~~~~~~~
/home/strife/.local/opt/zig/lib/std/mem.zig:4216:5: note: function declared here
pub fn alignForward(comptime T: type, addr: T, alignment: T) T {
~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
referenced by:
    decode__anon_70774: src/analyser/encoding.zig:134:35
    decode__anon_61744: src/analyser/encoding.zig:163:54
    remaining reference traces hidden; use '-freference-trace' to see all reference traces
```
The origin of this error can be directly attributed to [commit d41111d7ef531f6f55a19c56205d6d2f1134c224 in Zig, `mem: rename align*Generic to mem.align*`](https://github.com/ziglang/zig/commit/d41111d7ef531f6f55a19c56205d6d2f1134c224), committed less than a day ago. The obvious course of action to fix this (tested, works; should I just make a pull request?) would be to change the call to `std.mem.alignForward` to pass the alignment's type – in this case `usize` – as per the new definition, formerly known as `std.mem.alignForwardGeneric`.

--- a/build.zig
+++ b/build.zig
@@ -2,12 +2,12 @@ const std = @import("std");
 const builtin = @import("builtin");
 const shared = @import("src/shared.zig");
 
-const zls_version = std.builtin.Version{ .major = 0, .minor = 11, .patch = 0 };
+const zls_version = std.SemanticVersion{ .major = 0, .minor = 11, .patch = 0 };
 
 pub fn build(b: *std.build.Builder) !void {
     comptime {
         const current_zig = builtin.zig_version;
-        const min_zig = std.SemanticVersion.parse("0.11.0-dev.3312+ab37ab33c") catch unreachable; // poly1305: properly cast the mask from u1 to u64
+        const min_zig = std.SemanticVersion.parse("0.11.0-dev.3696+8d0a8c285") catch unreachable; // std.builtin.Version -> std.SemanticVersion
         if (current_zig.order(min_zig) == .lt) {
             @compileError(std.fmt.comptimePrint("Your Zig version v{} does not meet the minimum build requirement of v{}", .{ current_zig, min_zig }));
         }
@@ -65,7 +65,7 @@ pub fn build(b: *std.build.Builder) !void {
                 const commit_height = it.next().?;
                 const commit_id = it.next().?;
 
-                const ancestor_ver = try std.builtin.Version.parse(tagged_ancestor);
+                const ancestor_ver = try std.SemanticVersion.parse(tagged_ancestor);
                 std.debug.assert(zls_version.order(ancestor_ver) == .gt); // zls version must be greater than its previous version
                 std.debug.assert(std.mem.startsWith(u8, commit_id, "g")); // commit hash is prefixed with a 'g'
 

--- a/src/analyser/encoding.zig
+++ b/src/analyser/encoding.zig
@@ -193,7 +193,7 @@ pub fn canEncodeAsBytes(comptime T: type) bool {
 /// forward aligns `extra` until it has the given alignment
 pub fn alignForward(extra: []const u8, alignment: usize) []const u8 {
     const unaligned = @ptrToInt(extra.ptr);
-    const offset = std.mem.alignForward(unaligned, alignment) - unaligned;
+    const offset = std.mem.alignForward(usize, unaligned, alignment) - unaligned;
     const result = extra[offset..];
     std.debug.assert(std.mem.isAligned(@ptrToInt(result.ptr), alignment));
     return result;

